atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ReadTransaction.java

```diff
@@ -138,7 +138,8 @@ public Iterator<Map.Entry<Cell, byte[]>> getRowsColumnRange(
     private void checkTableName(TableReference tableRef) {
         SweepStrategy sweepStrategy = sweepStrategies.get(tableRef);
         Preconditions.checkState(
-                !sweepStrategy.mustCheckImmutableLockAfterReads(),
+                !sweepStrategy.mustCheckImmutableLock(true /* no read was actually done, but we don't want to use
+                ReadTransaction on thoroughly swept tables */),
```

Comment:
this is correct, if nasty.

unclear/preference: Does it even makes sense to have `mayHaveToCheckImmutableLock()` which calls the `||` of the two cases for this one?

Reply:
Probably not. I'll just switch to explicitly checking the sweep strategy